### PR TITLE
Update delta.chat

### PR DESCRIPTION
**Pull request by r10s**
only comparing for `zh` gives "Simplified Chinese" as used China also to ppl from Taiwan - but Taiwan uses "Traditional Chinese". to not offend ppl, better stay with English in this case (as before)

### CODE REVIEW

File: delta.chat/delta.chat [hunk 1]
```diff
@@ -32,8 +32,6 @@ map $http_accept_language $lang {
      ~ru ru;
      ~sq sq;
      ~uk uk;
-     ~zh-rCN zh-CN;
-     ~zh-Hans zh-CN;
```
Comment (r10s) [inline delta.chat/delta.chat:36]: i assume, this comes from my "fun fact" at https://github.com/deltachat/deltachat-pages/issues/618#issuecomment-1437439954 - but that was not meant to be used in the browsers, the languages codes inside the browser world is quite stable afaik - it is only different from what apps and different os outside browsers are doing. sorry for being unclear before.

File: delta.chat/delta.chat [hunk 2]
```diff
@@ -117,7 +115,7 @@ server {
                     for supported, translated in pairs(supported) do
                         if string.sub(lang, 0, 2) == supported then
  		            ngx.redirect("https://delta.chat/"..supported..ngx.var.request_uri, 301)
-			elseif string.sub(lang, 0, 2) == "zh" then
+			elseif string.sub(lang, 0, 2) == "zh-CN" then
```
Comment (r10s) [inline delta.chat/delta.chat:118]:
```suggestion
			elseif string.sub(lang, 0, 5) == "zh-CN" then
```

File: delta.chat/delta.chat (hunk 1 above)
Comment (missytake) [reply delta.chat/delta.chat:36]: it was taken from that comment, indeed :) I interpreted it as "people who have chinese configured on their mobile phones have their browsers sending out `zh-rCN` or `zh-Hans` in the Accept-Language header". This isn't the case?

Comment (r10s) [reply delta.chat/delta.chat:36]: > This isn't the case?

no. in browsers it is always  zh-CN. it is different when programming against the operating systems.

**Comment by missytake** [comment]
Hm, `zh` doesn't work anymore with the new config. Do we want it to redirect to `zh_CN` or not? It makes some sense... We could create a special elseif for `zh`.

**Comment by missytake** [comment]
Ah, re-read your top comment. Yes, let's show english in these cases. They can still select simplified chinese at the bottom (and hopefully find it).

I added tests for this so the expected behavior is made explicit in these cases.

**Comment by gerryfrancis** [comment]
I still get a redirection error in Firefox (for Desktop and Android) when I enter https://delta.chat/se in the address bar, so I believe that this issue still exists.

**Comment by r10s** [comment]
> I still get a redirection error in Firefox (for Desktop and Android) when I enter https://delta.chat/se in the address bar, so I believe that this issue still exists. (Caches have been cleared before.)

the 404 is on purpose and explained in detail at https://github.com/deltachat/deltachat-pages/issues/618#issuecomment-1438339359

**Pull request merged**